neanderthal/src/main/kotlin/au/com/outware/neanderthal/presentation/presenter/VariantListPresenter.kt

```diff
@@ -69,16 +77,47 @@ class VariantListPresenter @Inject constructor(): Presenter {
     }
 
     fun onEditClicked() {
-        view.goToEditVariant(currentVariantName)
+        if(variants.size > 0) {
+            view.goToEditVariant(currentVariantName)
+        }
     }
 
     fun onDeleteClicked() {
-        view.createDeleteConfirmation()
+        if(variants.size > 0) {
+            view.createDeleteConfirmation()
+        }
     }
 
     fun onAddVariant(name: String) {
         currentVariantName = name
-        adapter.add(name)
+        variants.add(name)
+        variants.sort()
+        adapter.add(variants)
+    }
+
+    fun onResetToDefaultClicked(){
+        view.createResetConfirmation()
+    }
+
+    fun onResetConfirmation(confirmed: Boolean){
+        if(confirmed) {
+            variantInteractor.resetVariants()
+            variants.clear()
+
+            val currentVariants : List<String> = variantInteractor.getVariantNames()
+            for(variant in currentVariants){
+                variants.add(variant)
+            }
+
+            variants.sort()
+            adapter.add(variants)
+            currentPosition = 0
+            currentVariantName = variants[0]
```

Comment:
taken from onCreate, use variantInteractor.getCurrentVariant()?.name ?: variants.first() so that we will be using what the initial default variant was, instead of whatever is first in the list